addons/website_sale/views/product_product_add.xml

```diff
@@ -20,6 +24,7 @@
         <field name="res_model">product.product</field>
         <field name="view_mode">form</field>
         <field name="target">new</field>
+        <field name="context" eval="{'default_available_in_pos': True, 'dialog_size': 'medium', 'isEcommerce': True}"/>
```

Comment:
Are you sure about that?
We don't want to make products created from eCommerce available in PoS by default :thinking:

Reply:
As discussion with @mobt-odoo he suggested to make `available_in_pos` as default true in all the form views of all modules. actually this idea was proposed by FP.

Reply:
If the objective is to make it true by default in all scopes, why not directly on the field itself? :thinking: 
Also, what is this `isEcommerce` key used for ?

Reply:
The `default="true"` attribute was already specified, but it was not reflecting on the UI. To address this, I have added default="true" in the context.

The `isEcommerce` field is used to display related categories in the view. Previously, the form view was showing only POS categories, which is irrelevant in the eCommerce module. To resolve this, we have included eCommerce categories from website_sale and ensured that the isEcommerce context is present to confirm the eCommerce module is installed. As a result, the POS categories will be hidden in this form view, with the form view of website_sale set as primary, so pos_categories will be hide from this perticular form view.

Please review and suggest any improvements wherever i missed something.

Reply:
> The `default="true"` attribute was already specified in the field, but it was not reflecting on the UI. To address this, I have added default="true" in the context.

What do you mean by 'in the field' ?
`available_in_pos = fields.Boolean(default=True)` ?
If that's it, it's strange that it doesn't work :thinking: 

> The `isEcommerce` field is used to display related categories in the view. Previously, the form view was showing only POS categories, which is irrelevant in the eCommerce module. To resolve this, we have included eCommerce categories from website_sale and ensured that the isEcommerce context is present to confirm the eCommerce module is installed. As a result, the POS categories will be hidden in this (website_sale) form view, with the form view of website_sale set as primary, so pos_categories will be hide from this (website_sale) perticular form view.

Why would you need that if the views are `primary` ?
Maybe you should simply set the PoS view to `primary` too, to ensure that both the Pos and the Ecommerce views do not impact each other and are only shown in the specific action for which they're expected to appear ?

Reply:
How about making one primary view (holding the pos categories field), and one non primary view in `point_of_sale` ?
That way, you wouldn't need the `isPoS` and `isEcommerce` context keys

Reply:
Actually we have this form view in 3 modules
1) Point_of_sale
2) Website_sale
3) stock_barcode
Now we want to show `available_in_pos` field in both modules `point_of_sale` and `stock_barcode` so we can't make point_of_sale view as primary as we want those fields in stock_barcode.
But at the same time we want category to be more precise and according to module so overriding the category here in e_commerce module only by making it primary and hiding only at this case.
If we create primary pos view than we won't get available_in_pos in any other module.

So briefly,

-----------------------------------|--point_of_sale view---|---stock_barcode----|---website_sale---
pos_categ---------------------|------------Yes-------------|----------Yes------------|--------No----------
eCommerce Categories---|-------------No-------------|----------No-------------|-------Yes----------
Barcode widget--------------|------------Yes-------------|----------No-------------|-------Yes-----------

Reply:
Yeah, but if you have two views in pos, wouldn't you be able to solve it more cleanly without the context keys ?
View 1: holds the `available_in_pos` field, not primary
View 2: primary, holds the pos categories field, extends View 1, and is used by the Pos product creation action.

Reply:
Yes, you are right it will work perfectly.
I have changed it accordingly, please check it again thanks.